In [2]:
#import libraries
import numpy as np
import pandas as pd

### 1.0 Data Collection

In [3]:
#read csv file
df = pd.read_csv("./1. Weather Data.csv")

#display first 10 rows
df.head(10)

Date/Time  Temp_C  Dew Point Temp_C  Rel Hum_%  Wind Speed_km/h  \
0  1/1/2012 0:00    -1.8              -3.9         86                4   
1  1/1/2012 1:00    -1.8              -3.7         87                4   
2  1/1/2012 2:00    -1.8              -3.4         89                7   
3  1/1/2012 3:00    -1.5              -3.2         88                6   
4  1/1/2012 4:00    -1.5              -3.3         88                7   
5  1/1/2012 5:00    -1.4              -3.3         87                9   
6  1/1/2012 6:00    -1.5              -3.1         89                7   
7  1/1/2012 7:00    -1.4              -3.6         85                7   
8  1/1/2012 8:00    -1.4              -3.6         85                9   
9  1/1/2012 9:00    -1.3              -3.1         88               15   

   Visibility_km  Press_kPa               Weather  
0            8.0     101.24                   Fog  
1            8.0     101.24                   Fog  
2            4.0     101.26  Freezing Drizzle,Fog  
3            4.0     101.27  Freezing Drizzle,Fog  
4            4.8     101.23                   Fog  
5            6.4     101.27                   Fog  
6            6.4     101.29                   Fog  
7            8.0     101.26                   Fog  
8            8.0     101.23                   Fog  
9            4.0     101.20                   Fog

In [4]:
#display last 10 rows
df.tail(10)

Date/Time  Temp_C  Dew Point Temp_C  Rel Hum_%  Wind Speed_km/h  \
8774  12/31/2012 14:00    -3.4              -5.7         84                6   
8775  12/31/2012 15:00    -2.3              -4.6         84                9   
8776  12/31/2012 16:00    -1.4              -4.0         82               13   
8777  12/31/2012 17:00    -1.1              -3.3         85               19   
8778  12/31/2012 18:00    -1.3              -3.1         88               17   
8779  12/31/2012 19:00     0.1              -2.7         81               30   
8780  12/31/2012 20:00     0.2              -2.4         83               24   
8781  12/31/2012 21:00    -0.5              -1.5         93               28   
8782  12/31/2012 22:00    -0.2              -1.8         89               28   
8783  12/31/2012 23:00     0.0              -2.1         86               30   

      Visibility_km  Press_kPa Weather  
8774           11.3     100.57    Snow  
8775            9.7     100.47    Snow  
8776           12.9     100.40    Snow  
8777            9.7     100.30    Snow  
8778            9.7     100.19    Snow  
8779            9.7     100.13    Snow  
8780            9.7     100.03    Snow  
8781            4.8      99.95    Snow  
8782            9.7      99.91    Snow  
8783           11.3      99.89    Snow

### 2.0 Shape of the Dataset

In [5]:
#get the shape
print(f"The dataframe has {df.shape[0]} rows and {df.shape[1]} columns")

The dataframe has 8784 rows and 8 columns


### 3.0 Find Records where Weather Is Clear

In [6]:
#get weather is clear
df[df['Weather'] == 'Clear']

Date/Time  Temp_C  Dew Point Temp_C  Rel Hum_%  Wind Speed_km/h  \
67      1/3/2012 19:00   -16.9             -24.8         50               24   
114     1/5/2012 18:00    -7.1             -14.4         56               11   
115     1/5/2012 19:00    -9.2             -15.4         61                7   
116     1/5/2012 20:00    -9.8             -15.7         62                9   
117     1/5/2012 21:00    -9.0             -14.8         63               13   
...                ...     ...               ...        ...              ...   
8646   12/26/2012 6:00   -13.4             -14.8         89                4   
8698  12/28/2012 10:00    -6.1              -8.6         82               19   
8713   12/29/2012 1:00   -11.9             -13.6         87               11   
8714   12/29/2012 2:00   -11.8             -13.1         90               13   
8756  12/30/2012 20:00   -13.8             -16.5         80               24   

      Visibility_km  Press_kPa Weather  
67             25.0     101.74   Clear  
114            25.0     100.71   Clear  
115            25.0     100.80   Clear  
116            25.0     100.83   Clear  
117            25.0     100.83   Clear  
...             ...        ...     ...  
8646           25.0     102.47   Clear  
8698           24.1     101.27   Clear  
8713           25.0     101.31   Clear  
8714           25.0     101.33   Clear  
8756           25.0     101.52   Clear  

[1326 rows x 8 columns]

- The records are 1326 where the weather was clear

### 4.0 Count Number of Times wind speed was exactly 4km/hr

In [9]:
#loc wind speed
wind_speed = df[df['Wind Speed_km/h'] == 4]
print(f"The number of times wind speed was exactly 4km/hr is {wind_speed.shape[0]} times")

The number of times wind speed was exactly 4km/hr is 474 times


### 5.0 Check Null values

In [10]:
#check for null values
df.isnull().sum()

Date/Time           0
Temp_C              0
Dew Point Temp_C    0
Rel Hum_%           0
Wind Speed_km/h     0
Visibility_km       0
Press_kPa           0
Weather             0
dtype: int64

### 6.0 Rename Weather column

In [14]:
#rename column
df = df.rename(columns={'Weather': 'Weather_Condition'})

#check if it has changed
df.columns

Index(['Date/Time', 'Temp_C', 'Dew Point Temp_C', 'Rel Hum_%',
       'Wind Speed_km/h', 'Visibility_km', 'Press_kPa', 'Weather_Condition'],
      dtype='object')

### 7.0 Check Mean Visibility


In [17]:
#use mean function
df['Visibility_km'].mean()

print(f"The mean visibility is {round(df['Visibility_km'].mean(), 3)}")

The mean visibility is 27.664


### 8.0 Wind Speed and Visibility

In [23]:
#check for records where wind speed is greater than 24km/hr
wind_speed_and_visibilty = df[(df['Wind Speed_km/h'] > 24) & (df['Visibility_km'] == 25)]
wind_speed_and_visibilty.shape[0]
print(f"The number of records are {wind_speed_and_visibilty.shape[0]}")

The number of records are 308


### 9.0 Mean Value for Each Weather Condition

In [29]:
#mean value for each weather condition
weather_count = df['Weather_Condition'].value_counts()
weather_mean = weather_count.mean()
weather_mean

175.68

### 10. Instances of Clear Weather and Relative Humidity and Visibility

In [34]:
#check
clear_weather_and_visibilty = df[(df['Weather_Condition'] > 'Clear') & (df['Rel Hum_%'] > 50) | (df['Visibility_km'] > 40) ]
clear_weather_and_visibilty


Date/Time  Temp_C  Dew Point Temp_C  Rel Hum_%  Wind Speed_km/h  \
0        1/1/2012 0:00    -1.8              -3.9         86                4   
1        1/1/2012 1:00    -1.8              -3.7         87                4   
2        1/1/2012 2:00    -1.8              -3.4         89                7   
3        1/1/2012 3:00    -1.5              -3.2         88                6   
4        1/1/2012 4:00    -1.5              -3.3         88                7   
...                ...     ...               ...        ...              ...   
8779  12/31/2012 19:00     0.1              -2.7         81               30   
8780  12/31/2012 20:00     0.2              -2.4         83               24   
8781  12/31/2012 21:00    -0.5              -1.5         93               28   
8782  12/31/2012 22:00    -0.2              -1.8         89               28   
8783  12/31/2012 23:00     0.0              -2.1         86               30   

      Visibility_km  Press_kPa     Weather_Condition  
0               8.0     101.24                   Fog  
1               8.0     101.24                   Fog  
2               4.0     101.26  Freezing Drizzle,Fog  
3               4.0     101.27  Freezing Drizzle,Fog  
4               4.8     101.23                   Fog  
...             ...        ...                   ...  
8779            9.7     100.13                  Snow  
8780            9.7     100.03                  Snow  
8781            4.8      99.95                  Snow  
8782            9.7      99.91                  Snow  
8783           11.3      99.89                  Snow  

[7377 rows x 8 columns]

### 11. Number of Weather Conditions that include snow

In [38]:
#number that include snow
snow_count = df[df['Weather_Condition'] == 'Snow']
print(f"The number of weather conditions that include snow are {snow_count.shape[0]}.")

The number of weather conditions that include snow are 390.
